### ISSUE TITLE
TypeError: "listener" argument must be a function

### ISSUE THREAD

**Issue by fishcharlie** [jamhall/s3rver#438]
I'm getting a bunch of errors like the following when running `npm test`

```
  1) S3rver Class Tests
       should support running on port 0:
     TypeError: "listener" argument must be a function
      at new Server (_http_server.js:269:10)
      at Object.createServer (http.js:34:10)
      at S3rver.listen (lib/s3rver.js:148:8)
      at runAsync (lib/s3rver.js:130:36)
      at <anonymous>
      at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

**Comment by kherock** [comment]
This looks like it might be solved by #434. Are you running Node < 8.12?

**Comment by fishcharlie** [comment]
@kherock Yes, Node.js 8.10

**Comment by kherock** [comment]
This has been fixed in v3.0.3!
